src/main/java/io/pravega/connectors/flink/PravegaInputSplit.java

```diff
@@ -40,7 +41,8 @@ public SegmentRange getSegmentRange() {
     }
 
     // --------------------------------------------------------------------
-
+    // cconstructor guards segment range from being null
+    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_MIGHT_BE_INFEASIBLE")
```

Comment:
Why the warning is suppressed?

Reply:
There was only one constructor and it looked like it guarded against nulls there. I debated on putting the checks in the equals method or  relying on the constructor constraints and putting the suppress in. I'm equally fine with either way. In hindsight, it's possible to change it after construction. I can put the guards  in the method instead.

Reply:
Added the guard into the source code and removed the suppression annotation